### PR TITLE
Spark: Add missing deprecation annotations for old actions

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR adds missing deprecation annotations for old Spark actions.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
@kbendick @karuppayya @RussellSpitzer @rdblue @flyrain, could you review, please?

**Review by RussellSpitzer — APPROVED** [review]

**Review by rdblue — APPROVED** [review]

**Comment by flyrain** [comment]
+1 for the PR.

**Comment by aokolnychyi** [comment]
This failed as we need to get #2810 first.

**Review by kbendick — APPROVED** [review]
LGTM.

**Comment by aokolnychyi** [comment]
Thanks, @kbendick @rdblue @flyrain @RussellSpitzer!

File: spark/src/main/java/org/apache/iceberg/actions/RewriteDataFilesAction.java [hunk 1]
```diff
@@ -32,6 +32,10 @@
 import org.apache.spark.broadcast.Broadcast;
 import org.apache.spark.sql.SparkSession;
 
+/**
+ * @deprecated since 0.12.0, will be removed in 0.13.0; use {@link RewriteDataFilesAction} instead.
+ */
+@Deprecated
```
Comment (jshmchenxi) [inline spark/src/main/java/org/apache/iceberg/actions/RewriteDataFilesAction.java:38]: Hi, should we use `BaseRewriteDataFilesAction` instead? `RewriteDataFilesAction` is the deprecated class itself.

**Pull request merged**